generator/program.go

```diff
@@ -0,0 +1,389 @@
+const (
+	ExitCodeOkay int = iota
+	ExitCodeMissingRequirements
+	ExitCodeFailedToClone
+)
+
+var (
+	remoteAzureRestAPISpecsPath string
+	localAzureRestAPISpecsPath  string
+	azureRestAPIBranch          string
+	outputLocation              string
+	dryRun                      bool
+	help                        bool
+	anyMissing                  bool
+	noClone                     bool
+	verbose                     bool
+	wait                        bool
+	targetFile                  string
+	debugLog                    *log.Logger
+	version                     *semver.Version
+)
+
+type Swagger struct {
+	swagger.Swagger
+	Path string
+}
+
+func init() {
+	var err error
+	flag.StringVar(&azureRestAPIBranch, "branch", defaultAzureRESTAPIBranch, "The branch, tag, or SHA1 identifier in the Azure Rest API Specs repository to use during API generation.")
+	flag.StringVar(&remoteAzureRestAPISpecsPath, "repo", defaultRemoteAzureRestAPISpecsPath, "The path to the location of the Azure REST API Specs repository that should be used for generation.")
+	flag.StringVar(&outputLocation, "output", getDefaultOutputLocation(), "a directory in which all generated code should be placed.")
+	flag.StringVar(&targetFile, "target", "", "If a target file is provided, generator will only run on this file instead of all swaggers it encounters in the repository.")
+	flag.BoolVar(&dryRun, "preview", false, "Use this flag to print a list of packages that would be generated instead of actually generating the new sdk.")
+	flag.BoolVar(&help, "help", false, "Provide this flag to print usage information instead of running the program.")
+	flag.BoolVar(&noClone, "noClone", false, "Use this flag to prevent cloning a new copy of Azure REST API Specs. The existing enlistment should be used instead.")
+	flag.BoolVar(&verbose, "verbose", false, "Print status messages as processing is done.")
+	flag.BoolVar(&wait, "wait", false, "Use this program to halt execution before the cleanup phase is entered.")
+	useDebug := flag.Bool("debug", false, "Include this flag to print debug messages as the program executes.")
+	rawVersion := flag.String("version", "0.0.0", "The version that should be stamped on the generated code. Common usage will be to report user agent.")
+
+	flag.Parse()
+
+	if help {
+		return
+	}
+
+	version, err = semver.NewVersion(*rawVersion)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Could not read version \"%s\" because: %v", *rawVersion, err)
+		os.Exit(1)
+	}
+
+	debugWriter := ioutil.Discard
+	if *useDebug {
+		fmt.Fprintln(os.Stdout, "Turning on debug logger.")
+		debugWriter = os.Stderr
+	}
+	debugLog = log.New(debugWriter, "[DEBUG]", log.Ltime)
+
+	optionalTools := []string{"gofmt", "golint"}
+	requiredTools := []string{"autorest", "git"}
+
+	for _, tool := range optionalTools {
+		if _, err := exec.LookPath(tool); err != nil {
+			log.Printf("WARNING: Could not find \"%s\" usage of this tool will be skipped.", tool)
+		}
+	}
+
+	anyMissing = false
```

Comment:
Right you are! Personally, I like to set these things explicitly for readability and just in case it gets set somewhere else.